R/driver-db2.R

```diff
@@ -29,3 +30,44 @@ setMethod("sqlCreateTable", "DB2/AIX64",
     ))
   }
 )
+
+
+#' @rdname DBI-methods
+# In addition to the `ODBC` API call to `SQLTables`, also
+# attempt to query SYSIBMADM.ADMINTEMPTABLES for any tables
+# that match the arguments.
+#' @usage NULL
+setMethod("odbcConnectionTables", c("DB2/AIX64", "character"),
+  function(conn,
+           name,
+           catalog_name = NULL,
+           schema_name = NULL,
+           table_type = NULL,
+           exact = FALSE) {
+
+    res <- callNextMethod()
+    dfTempTables <- NULL
+    if ( (is.null(schema_name) || schema_name == "SESSION") &&
+         (is.null(catalog_name) || tolower(catalog_name) == conn@info$username) ) {
+      # Also look for temp tables in this case
+      qTable <- getSelector("TABNAME", name, exact)
+      qSchema <- getSelector("TABSCHEMA", schema_name, exact)
+      query <- paste0(
+        "SELECT
+            INSTANTIATOR AS \"table_catalog\",
+            TABSCHEMA AS \"table_schema\",
+            TABNAME AS \"table_name\",
+            'TABLE' AS \"table_type\",
+            NULL AS \"table_remarks\"
+         FROM SYSIBMADM.ADMINTEMPTABLES ",
+        "WHERE 1 = 1 ", qTable, qSchema)
+      tryCatch({
```

Comment:
Why do we need a `tryCatch` here? Could we scope it more narrowly to a subclass of error?

Reply:
I am no `DB2` expert, but I was a bit concerned that not every user might have access to tables/views prefixed with `ADMIN`.

Not sure if the approach I took in the follow up is what you had in mind - subclass `rethrow_database_error::cli_abort` - let me know if not.